app/models/miq_widget_set.rb

```diff
@@ -14,6 +33,33 @@ def self.with_users
     where.not(:userid => nil)
   end
 
+  def update_members
+    replace_children(Array(set_data_widgets))
+    current_user = User.current_user
+    members.each { |w| w.create_initial_content_for_user(current_user.userid) } if current_user # Generate content if not there
+  end
+
+  def add_to_dashboard_order
+    return unless group_id
+
+    group = MiqGroup.find(group_id)
+    group.add_to_dashboard_order(id)
+    group.save
+  end
+
+  def delete_from_dashboard_order
+    return unless group_id
+
+    group = MiqGroup.find(group_id)
+    group.delete_from_dashboard_order(id)
+    group.save
+  end
+
+  def ensure_can_be_destroyed
+    errors.add(:base, _("Unable to delete read-only WidgetSet")) if read_only?
+    throw :abort unless errors[:base].empty?
```

Comment:
is this trying to catch errors not added by this class?

or should this all go into a block?

```suggestion
    if read_only?
      errors.add(:base, _("Unable to delete read-only WidgetSet"))
      throw :abort
    end
```

Reply:
it looks that `throw :abort` is not needed here (I don't see `errors.add(:base)` on other places)